keel-igor/src/test/kotlin/com/netflix/spinnaker/keel/services/BuildLifecycleMonitorTests.kt

```diff
@@ -0,0 +1,312 @@
+    )
+
+    val task = MonitoredTask(
+      link = id,
+      triggeringEvent = event
+    )
+
+    val subject = BuildLifecycleMonitor(
+      monitorRepository,
+      publisher,
+      lifecycleConfig,
+      objectMapper,
+      artifactMetadataService,
+      front50Service,
+      spinnakerBaseUrl
+    )
+  }
+
+  fun tests() = rootContext<Fixture> {
+    fixture { Fixture() }
+
+    test("handles builds") {
+      expectThat(subject.handles(BUILD)).isTrue()
+    }
+
+    context("updating status") {
+      before {
+        coEvery { front50Service.applicationByName(any()) } returns app
+      }
+      context("building") {
+        before {
+          coEvery { artifactMetadataService.getArtifactMetadata(buildNumber, commitId) } returns getArtifactMetadata("BUILDING")
+          runBlocking { subject.monitor(task) }
+        }
+
+        test("emits running event") {
+          val slot = slot<LifecycleEvent>()
+          verify(exactly = 1) {
+            publisher.publishEvent(capture(slot))
+          }
+          expectThat(slot.captured.status).isEqualTo(RUNNING)
+        }
+      }
+
+      context("success") {
+        before {
+          coEvery { artifactMetadataService.getArtifactMetadata(buildNumber, commitId) } returns
+            getArtifactMetadata("SUCCESS")
+          runBlocking { subject.monitor(task) }
+        }
+
+        test("emits succeeded event") {
+          val slot = slot<LifecycleEvent>()
+          verify(exactly = 1) {
+            publisher.publishEvent(capture(slot))
+          }
+          expectThat(slot.captured.status).isEqualTo(SUCCEEDED)
+        }
+      }
+
+      context("failure") {
+        before {
+          coEvery { artifactMetadataService.getArtifactMetadata(buildNumber, commitId) } returns
+            getArtifactMetadata("FAILURE")
+          runBlocking { subject.monitor(task) }
+        }
+
+        test("emits failed event") {
+          val slot = slot<LifecycleEvent>()
+          verify(exactly = 1) {
+            publisher.publishEvent(capture(slot))
+          }
+          expectThat(slot.captured.status).isEqualTo(FAILED)
+        }
+      }
+
+      context("aborted") {
+        before {
+          coEvery { artifactMetadataService.getArtifactMetadata(buildNumber, commitId) } returns
+            getArtifactMetadata("ABORTED")
+          runBlocking { subject.monitor(task) }
+        }
+
+        test("emits aborted event") {
+          val slot = slot<LifecycleEvent>()
+          verify(exactly = 1) {
+            publisher.publishEvent(capture(slot))
+          }
+          expectThat(slot.captured.status).isEqualTo(ABORTED)
+        }
+      }
+
+      context("UNSTABLE") {
+        before {
+          coEvery { artifactMetadataService.getArtifactMetadata(buildNumber, commitId) } returns
+            getArtifactMetadata("UNSTABLE")
+          runBlocking { subject.monitor(task) }
+        }
+
+        test("emits succeeded event") {
```

Comment:
Consider adding a comment here explaining why it is that we expect a succeeded event when the status is `UNSTABLE`.

Reply:
I'll add a comment in the build monitor itself